lib/shopwareStoreCommander.js

```diff
@@ -0,0 +1,364 @@
+const fs = require('fs');
+const EventEmitter = require('events');
+const FormData = require('form-data');
+const concat = require('concat-stream');
+const shopwareStoreClient = require('./client/shopwareStoreClient');
+
+function getPostData(filePath) {
+    return new Promise((resolve) => {
+        const fd = new FormData();
+        fd.append('file', fs.createReadStream(filePath));
+        fd.pipe(concat({ encoding: 'buffer' }, data => resolve({ data, headers: fd.getHeaders() })));
+    });
+}
+
+module.exports = class ShopwareStoreCommander {
+
+    /**
+     * @constructor
+     * @param {String} username
+     * @param {String} password
+     * @param {EventEmitter} logger - optional
+     */
+    constructor(username, password, logger = new EventEmitter()) {
+        this.client = shopwareStoreClient(username, password, logger);
+        this.logger = logger;
+        this.accountData = {};
+        this.statics = {};
+    }
+
+    /**
+     * @return {Promise}
+     */
+    async loadAccountData() {
+        const producers = await this.client.get('producers');
+
+        // Save producer ID (required to load e.g. plugins)
+        this.accountData = {
+            producerId: producers.data[0].id,
+        };
+
+        const plugins = await this.client.get('plugins', {
+            params: {
+                offset: 0,
+                limit: 1000,
+                producerId: this.accountData.producerId,
+            },
+        });
+
+        // Save plugins
+        this.accountData.plugins = {};
+        plugins.data.forEach((plugin) => {
+            this.accountData.plugins[plugin.name] = plugin;
+        });
+
+        // Load static definitions like error codes etc.
+        const pluginStatics = await this.client.get('pluginstatics/all');
+
+        this.statics = pluginStatics.data;
+
+        return this.accountData;
+    }
+
+    /**
+     * @param {String} pluginName
+     * @param {Object} extraFields - optional
+     * @return {Promise}
+     */
+    async findPlugin(pluginName, extraFields) {
+        const accountData = await ((!this.accountData.plugins)
```

Comment:
Here, you decide which method (with side effects) you run based on current state. This is imperative programming, so I'd recommend an imperative `if` instead of the ternary operator here. 

In general, I think code that `await`s the result of a complex expression is very hard to reason about, so `awaiting` a simple function call is usually better IMHO.